src/common/scratch_buffer.hpp

```diff
@@ -13,6 +15,26 @@
 /// TODO: use it for
 /// - SHA256 encoding
 
-#define SCRATCH_BUFFER_SIZE 49152 ///< ugly but constexpr int does not work
+struct ScratchBuffer {
+    uint8_t buffer[49152];
+
+    size_t size() const {
+        return sizeof(buffer);
+    }
+};
+
+class Ownership {
+private:
+    bool acquired;
+
+public:
+    Ownership();
```

Comment:
Is there a reason for not having this class a singleton? AFAIK CCMRAM area doesn't depend on any HW and can be inited any time - i.e. an ideal example for a singleton.

Reply:
There is a singleton – the ScratchBuffer instance (intentionally private in scratch_buffer.cpp).

This class represents the Ownership of the buffer (RAII). Thus it is not a singleton.

By making the singleton private, the API does not allow (without acquiring an ownership first) use of the buffer.

Reply:
ok, got it